src/Choices/Catalog.php

```diff
@@ -10,12 +10,40 @@
 
 namespace BitBag\SyliusCatalogPlugin\Choices;
 
-final class Catalog
+use Symfony\Component\Finder\Exception\DirectoryNotFoundException;
+use Symfony\Component\Finder\Finder;
+
+final class Catalog implements CatalogInterface
 {
-    public static function getTemplates(): array
+    private string $fullTemplatePath;
+
+    private CatalogMapperInterface $catalogMapper;
+
+    public function __construct(string $fullTemplatePath, CatalogMapperInterface $catalogMapper)
     {
-        return [
-            'bitbag_sylius_catalog_plugin.ui.form.catalog.default_template' => '@BitBagSyliusCatalogPlugin/Catalog/Templates/showProducts.html.twig',
-        ];
+        $this->fullTemplatePath = $fullTemplatePath;
+        $this->catalogMapper = $catalogMapper;
+    }
+
+    public function getTemplates(): array
+    {
+        $finder = new Finder();
```

Comment:
Can we get `Finder` from DI?

Reply:
[Here](https://stackoverflow.com/questions/26016812/why-i-cant-get-symfony-finder-like-a-service) is explanation, why I pass it by reference. Should I make my own service or leave it like this?

Reply:
You can still register it as your own service

```xml
<service id="bitbag_sylius_catelog_plugin.filesystem.finder" class="Symfony\Component\Finder\Finder"/>
```